platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/Problem.java

```diff
@@ -52,9 +52,18 @@ public interface Problem {
     String getDetails();
 
     /**
-     * Return the location data associated available for this problem.
+     * Returns the locations where the problem originated.
+     * <p>
+     * Might be empty if the origin is not known.
+     */
+    List<ProblemLocation> getOriginLocations();
+
+    /**
+     * Returns additional locations, which can help to understand the problem further.
+     * <p>
+     * Might be empty if there is no meaningful contextual information.
      */
-    List<ProblemLocation> getLocations();
+    List<ProblemLocation> getContextualLocations();
```

Comment:
❔ For my understanding, we currently don't add any contextual location to a problem, right? Do you already have a PR that does this?

Reply:
Not yet, we tackled this separation first.